portability-core/src/test/java/org/dataportabilityproject/serviceProviders/google/contacts/GoogleContactsServiceTest.java

```diff
@@ -0,0 +1,127 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dataportabilityproject.serviceProviders.google.contacts;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.api.services.people.v1.PeopleService;
+import com.google.api.services.people.v1.PeopleService.People.Connections;
+import com.google.api.services.people.v1.model.EmailAddress;
+import com.google.api.services.people.v1.model.FieldMetadata;
+import com.google.api.services.people.v1.model.ListConnectionsResponse;
+import com.google.api.services.people.v1.model.Name;
+import com.google.api.services.people.v1.model.Person;
+import ezvcard.VCard;
+import ezvcard.property.Email;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import org.dataportabilityproject.cloud.interfaces.JobDataCache;
+import org.dataportabilityproject.cloud.local.InMemoryJobDataCache;
+import org.dataportabilityproject.dataModels.ExportInformation;
+import org.dataportabilityproject.dataModels.contacts.ContactsModelWrapper;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GoogleContactsServiceTest {
+
+  private static final Logger logger = LoggerFactory.getLogger(GoogleContactsServiceTest.class);
+
+  private static final String FIRST_NAME = "Jane";
+  private static final String LAST_NAME = "Doe";
+  private static final String RESOURCE_NAME = "resource_name";
+  private static final FieldMetadata PRIMARY_FIELD_METADATA = new FieldMetadata().setPrimary(true);
+  private static final Name NAME = new Name()
+      .setGivenName(FIRST_NAME)
+      .setFamilyName(LAST_NAME)
+      .setMetadata(PRIMARY_FIELD_METADATA);
+  private static final Person PERSON = new Person()
+      .setNames(Collections.singletonList(NAME))
+      .setResourceName(RESOURCE_NAME);
+
+  private Connections.List connectionsListRequest;
+  private PeopleService peopleService;
+  private JobDataCache jobDataCache;
+  private GoogleContactsService contactsService;
+
+  @Before
+  public void setup() {
+    connectionsListRequest = mock(Connections.List.class);
+    peopleService = mock(PeopleService.class, Mockito.RETURNS_DEEP_STUBS);
+    jobDataCache = new InMemoryJobDataCache();
+    contactsService = new GoogleContactsService(peopleService, jobDataCache);
+  }
+
+  @Test
+  public void convertToVCardEmails() {
+    // Set up test
+    int numEmails = 1;
+    List<EmailAddress> googleEmails = new LinkedList<>();
+    List<Email> expectedVCardEmails = new LinkedList<>();
+    for (int i = 0; i < numEmails; i++) {
```

Comment:
are you planning on having numEmails be more than 1? if not can you get rid of the loop?